livesync_citadel/indico_livesync_citadel/backend.py

```diff
@@ -0,0 +1,111 @@
+        }
+
+    def get_jsondata(self, obj):
+        # TODO: should "schemas/indico" be another setting?
+        if isinstance(obj, Event):
+            event_data = EventSchema().dump(obj)
+            event_data['$schema'] = url_join(self.search_app, 'schemas/indico/events_v1.0.0.json')
+            return event_data, EntryType.event
+        elif isinstance(obj, Contribution):
+            contribution_data = ContributionSchema().dump(obj)
+            contribution_data['$schema'] = url_join(self.search_app, 'schemas/indico/contributions_v1.0.0.json')
+            return contribution_data, EntryType.contribution
+        elif isinstance(obj, SubContribution):
+            subcontrib_data = SubContributionSchema().dump(obj)
+            subcontrib_data['$schema'] = url_join(self.search_app, 'schemas/indico/subcontributions_v1.0.0.json')
+            return subcontrib_data, EntryType.subcontribution
+        elif isinstance(obj, Attachment):
+            attachment_data = AttachmentSchema().dump(obj)
+            attachment_data['$schema'] = url_join(self.search_app, 'schemas/indico/attachments_v1.0.0.json')
+            return attachment_data, EntryType.attachment
+        elif isinstance(obj, EventNote):
+            note_data = EventNoteSchema().dump(obj)
+            note_data['$schema'] = url_join(self.search_app, 'schemas/indico/notes_v1.0.0.json')
+            return note_data, EntryType.note
+        else:
+            raise ValueError('unknown object ref: {}'.format(obj))
+
+    def upload_jsondata(self, jsondata, change_type, obj_id, entry_type):
+        response = None
+        if change_type & SimpleChange.created:
+            response = requests.post(self.endpoint_url, headers=self.headers, json=jsondata, verify=False)
+        else:
+            search_id = LiveSyncCitadelSearchAppIdMap.get_search_id(obj_id, entry_type)
+            if search_id:
+                if change_type & SimpleChange.updated:
+                    url = url_join(self.search_app, 'api/record/{}'.format(search_id))
+                    response = requests.put(url, headers=self.headers, json=jsondata, verify=False)
+                elif change_type & SimpleChange.deleted:
+                    url = url_join(self.search_app, 'api/record/{}'.format(search_id))
+                    response = requests.delete(url, headers=self.headers, json=jsondata, verify=False)
+            else:
+                raise LiveSyncCitadelUploaderError('SearchMapId does not exist for the object')
+
+        if not response.ok:
+            raise LiveSyncCitadelUploaderError('{} - {}'.format(response.status_code, response.text))
+        elif change_type & SimpleChange.created:
+            content = json.loads(response.content)
+            if content['metadata']['control_number']:
+                LiveSyncCitadelSearchAppIdMap.create(content['metadata']['control_number'], obj_id, entry_type)
```

Comment:
i think we may need to commit here: if we fail during a later object we don't want to lose all the IDs of already-stored objects...